setting download options - adding new art

gusreiber opened the pull request:
This PR still has a handful of issue to be resolved. I still have quite a bit of layout balancing to do, consideration for pages not the homepage, scroll position behaviors for menus and background positions, so still stuff to do, but giving an over-the-shoulder view as I go.

orrc commented:
Nice.  The download popup works for me, though the main download button doesn't seem to have rounded corners on the right-hand side, and the popup doesn't close if I click outside of it.

Also good is that the carousel is dead! :D  Though users on all types of devices have to load a 1.7MB video now?

orrc commented on content/_partials/downloadbanner.html.haml:
```diff
@@ -0,0 +1,58 @@
+        Your browser does not support the video tag.    
+      .mask
+      %div.carousel-caption.primary
+        %img.jenkins{:src=>"/images/226px-Jenkins_logo.svg.png"}
```
Should this image be part of this PR?

gusreiber replied:
The video may well be too heavy and may not be the right aesthetic. At the moment, I am tossing stuff into the soup and listening for comments.

orrc commented on content/_partials/downloadbanner.html.haml:
```diff
@@ -0,0 +1,58 @@
+        %img.jenkins{:src=>"/images/226px-Jenkins_logo.svg.png"}
+        %h1.page-title 
+          Jenkins 
+          %span.in-tag CD|CI
```
The name shouldn't include [CI or CD](https://jenkins-ci.org/press/#branding) :)

oleg-nenashev replied:
+1. Jenkins Automation Server

oleg-nenashev commented on content/css/jenkins.css:
```diff
@@ -26,7 +60,8 @@ div .authors {display:block; padding:0; margin-bottom:3rem}
 body .jumbotron {margin:0}
 body .card-block {display:block}
-.jumbotron.docker {box-shadow:inset 0 0 150px rgba(0,0,0,.3)}
+.jumbotron.docker {background:#042E54 url(/images/whale_1030x500.jpg) fixed no-repeat right}
```
Are we eligible to use a non-classic Docker image here? Could it cause trademark issues?

oleg-nenashev commented:
I would be happy to see a more detailed description of the PR and incoming changes. Seems the most of changes are unrelated to the summary

rtyler commented on content/_partials/downloadlist.html.haml:
```diff
@@ -1,42 +1,105 @@
+    .col-md-6.text-xs-center
+      %h3 LTS Release
+      %p.details
+        %a.item{:href=>'http://jenkins-ci.org/changelog'} Changelog
```
This should be relative

rtyler commented on content/_partials/downloadlist.html.haml:
```diff
@@ -1,42 +1,105 @@
+              %span.icon 
+              %span.title
+                Docker
+            %a.dropdown-item{:href=>'http://pkg.jenkins-ci.org/debian/'}
```
All these links will obviously need to be corrected. If that's immaterial to this change then you can safely ignore this comment

rtyler commented on content/_partials/downloadlist.html.haml:
```diff
@@ -1,42 +1,105 @@
+      %p
+        .btn-group
+          %a.btn.btn-primary{:href=>'http://mirrors.jenkins-ci.org/war/latest/jenkins.war'}
+            %i.icon-download  1.644 .war
```
FWIW there's a variable with the latest version which should be inserted here, `site.jenkins.latest`

rtyler commented on content/_partials/toptoolbar.html.haml:
```diff
@@ -13,7 +11,7 @@
 
   %a.navbar-brand{:href => '/'}
-    Jenkins
+    Jenkins.io
```
The branding needs to remain "Jenkins" wherever it goes.

rtyler commented on content/css/jenkins.css:
```diff
@@ -1,18 +1,52 @@
 #ji-download .dropdown-toggle {box-shadow:inset 0 0 0 200rem rgba(0,0,0,.2)}
 #ji-download .carousel-caption{ position:static; display:block; padding:1rem;}
 
+.carousel-item video {position:absolute; top:0; left:0; right:0; bottom:0; width:100%; opacity:.67
```
In the future, if CSS is being added, I would like to see some code-type comments to make sure that future contributors don't inadvertently break CSS when updating

rtyler commented:
@gusreiber Looking good, I've run these changes locally and here's some feedback on the overall view:

* (minor) when the download modal dialog comes up, clicking anywhere outside of the dialog should close it.
* echo @orrc about the background animation for the primary block on the home page
* also as @orrc mentioned, the Jenkins logo image asset referred to by the home page is not checked in.
* Events do not render correctly for me: ![jenkins-beta-events](https://cloud.githubusercontent.com/assets/26594/12297684/1c9cafce-b9c3-11e5-844d-bdcd0f5a76d9.png)
* From the navbar there is no longer the "Jenkins" title which a user can click to get back to the homepage: ![navbar-gohome](https://cloud.githubusercontent.com/assets/26594/12297722/4ea329bc-b9c3-11e5-8f66-3690da676f1a.png)
* The docker whale image either isn't appropriately sized for the space it's being used, or the CSS isn't correctly inserting it, making the whale's belly all that's visible on the frontpage. It stays fixed in the viewport when I scroll so I suspect the latter: ![horizontal-whale](https://cloud.githubusercontent.com/assets/26594/12297768/836a5cba-b9c3-11e5-9396-27061b703888.png)
* The blueish black theming doesn't match our colorscheme, though I'm not sure if that's supposed to be present in this changeset.


That's the first pass I'll do on this pull request, holler on IRC if you've got further questions

gusreiber commented:
some additional fixes, getting closer...

rtyler commented:
@gusreiber how mergable do you consider this right now? There's some some points I don't think the latest commit addresses (colortheme), and the new mystery meat navigation in the sidebar I'm really not fond of.

Should I consider this PR still a work in progress?

gusreiber commented:
For the color theme, I am not totally sure what the objection is. 
For the sidebar, I am thinking it will make sense for the social media links (tweet this), but yeah, otherwise should go, also hover effect isn't great, still a solution looking for a problem.

....so if it isn't clear, I am not totally sure how to answer your question, except that this PR is closer to done than is what is currently being shown on the beta.jenkins.io domain, so I would rather see this deployed. That way it will be easier to solicit feedback. Absent feedback, I am just going to continue addressing the issues that I see as the most glaring.

rtyler commented:
> For the color theme, I am not totally sure what the objection is. 

Not an objection so much as wondering if these colors are our final colors. We have a defined [color palette](https://wiki.jenkins-ci.org/display/JENKINS/Logo) which I strongly believe should be reflected in some form or fashion in the end-result of jenkins.io. With this pull request, it still appears that the site is using fairly "stock" bootstrap coloring.


Here's some notes on the latest set of commits:

1. `/images/226px-Jenkins_logo.svg.png` still isn't checked in properly. Which means we're all seeing a broken image in the header background on the front-page
1. The "Jenkins.io" title is wrong, it should only be "Jenkins" but that's really immaterial to this pull requests.
1. The top toolbar slide-in-from-the-right animation is wholly unnecessary IMO.
1. Clicking "Download Jenkins" shows the tooltip for the download (correct) but also shows the tooltips for all the "Getting started with Jenkins" boxes [see this screenshot](https://cloud.githubusercontent.com/assets/26594/12398244/33b53110-bdc7-11e5-8627-d76e98fa5192.png)
1. The footermenu partial looks promising, but unfortunately it looks like the "page" referenced for the "Improove this page" link is the partial itself, instead of the primary template being rendered. Wacky.
1. The new features boxes looks good
1. The mystery-meat sidebar thing, I grok that it's just some experimenting at right now, but doesn't deal with smaller screens so well and clips into blog content (FWIW)


As for deploying this, I don't see the benefit of deploying this pull request in its current form. We've had no problems generating feedback thus far, between you, @orrc and myself. I'd like to get things more cleaned up before punting out to beta.jenkins.io

gusreiber commented:
1, 2, 4, and 7 are fixed. I disagree about #3.
I am pretty sure there are at least a few people who feel this would be helpful up on the beta. url.

rtyler commented:
@gusreiber when will the colors of the site be on-brand? I'm fine merging this to get more feedback but I need to know when we're going to have our colors correct.

rtyler commented:
I should be more clear, the reason this is so important to me is that I've already had to field questions about "whether Jenkins is changing branding colors" based on the circulation of the `beta.jenkins.io`. It's a fairly important part of our branding

orrc commented:
Looks a lot better and feels much more complete than before! :sunglasses: 

I'm totally with Tyler on the animating toolbar things being unnecessary (which still says "Jenkins.io", it seems?), and I could live with 80% less LEGO®.

The various page sections look pretty good. Though I still see the background video on /blogs/?

gusreiber commented:
Sure. Just as the content isn't finished, the design isn't either. The
bricks are a little kitschie, so I may yet pull them out.

As an FYI, I won't have a ton of time this week, but will continue kicking
the can forward as I can.
On Jan 19, 2016 5:39 PM, "Christopher Orr" <notifications@github.com> wrote:

> Looks a lot better and feels much more complete than before! [image:
> :sunglasses:]
>
> I'm totally with Tyler on the animating toolbar things being unnecessary
> (which still says "Jenkins.io", it seems?), and I could live with 80% less
> LEGO®.
>
> The various page sections look pretty good. Though I still see the
> background video on /blogs/?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkins-infra/jenkins.io/pull/45#issuecomment-173055404>
> .
>

rtyler commented:
@gusreiber If I can find the time today, I'll rebase this pull request up and get it merged in and we can continue iterating.

gusreiber commented:
Appreciated. FWIW, my reasoning for the lego kitschiness is that having
some sort of theme device like that makes it easier to formulaically
generate art, which is often functionally useful to draw attention to
various bits on the page. Adherence to a red, white and blue color scheme
is somewhat helpful in that regard, but not a lot.

...so my question is, is my building brick treatment inherently too cheesy
or is it just over done at the moment? If it is too cheesy, does anyone
have an alternative suggestion that could help serve as a unifying thematic
device? If the concern is a use/rights issue, that is a false concern. I am
not doing anything here that infringes on the Lego TM.

For the colors of the Bootstrap buttons, I will want to attach a Less
compiler to the page generation so as to parameterize the colors rather
than specifying individual CSS rules. That is a bit of mechanics that is
tangential to actually figuring out what the pages should look like, so I
have been punting on it. I hear you, that punting is causing you pain, so I
will get to it.



On Wed, Jan 20, 2016 at 8:34 AM, R. Tyler Croy <notifications@github.com>
wrote:

> @gusreiber <https://github.com/gusreiber> If I can find the time today,
> I'll rebase this pull request up and get it merged in and we can continue
> iterating.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkins-infra/jenkins.io/pull/45#issuecomment-173263612>
> .
>
